### ISSUE TITLE
more than 1 pod => infinite loop and never deploy

### ISSUE THREAD

**Issue by flbla** [bohlenc/k8s-pod-mutator-webhook#5]
Hi @bohlenc 

if I deploy more than one pod, it goes in too infinite loop 
eg, I have those 2 deployments :
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment
  labels:
    app: nginx
spec:
  replicas: 1
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx:alpine
        ports:
        - containerPort: 80
```

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment2
  labels:
    app: nginx2
spec:
  replicas: 1
  selector:
    matchLabels:
      app: nginx2
  template:
    metadata:
      labels:
        app: nginx2
    spec:
      containers:
      - name: nginx2
        image: nginx:alpine
        ports:
        - containerPort: 80
```

with this path configuration :

```
webhook:
  patch: |
    spec:
      initContainers:
      - name: ca
        image: alpine
        command: ["sh", "-c", "cp -r /etc/ssl/certs /volume"]
        volumeMounts:
        - mountPath: /volume
          name: cacerts
      containers:
      - name: "*"
        volumeMounts:
        - mountPath: /etc/ssl
          name: cacerts
      volumes:
      - name: cacerts
        emptyDir: {}
```

I also tried with a named container instead of "*", same issue

**Comment by bohlenc** [comment]
Hi @flbla,

can you please describe the problem in more detail? Can you attach logs?

**Comment by flbla** [comment]
if you try to do exactly the same that I did,
The 1rst deployment will be OK, with the patch
and the 2nd will never deploy, trying infinitely to re-create the pod (it seems the mutator delete some fields like Image)

If then you delete the mutator and do exactly the same 2 nginx deployment it will work without any error.

**Comment by bohlenc** [comment]
Hi @flbla!
Sorry for the delay.
Fixed the bug with PR #6. Try out the new version 1.1.1!

**Comment by flbla** [comment]
Hi @bohlenc 
Thanks a lot!!!
I will test it :)
